src/software/thunderscope/robot_diagnostics/handheld_device_manager.py

```diff
@@ -0,0 +1,386 @@
+
+            if (
+                event.code
+                == self.controller_config[RobotControlType.DRIBBLER_SPEED][
+                    HandheldDeviceConfigKeys.CODE
+                ]
+            ):
+                self.dribbler_speed_accumulator = self.__parse_dribbler_event_value(
+                    event.value
+                )
+
+            if (
+                event.code
+                == self.controller_config[RobotControlType.DRIBBLER_ENABLE_1][
+                    HandheldDeviceConfigKeys.CODE
+                ]
+                or event.code
+                == self.controller_config[RobotControlType.DRIBBLER_ENABLE_2][
+                    HandheldDeviceConfigKeys.CODE
+                ]
+            ):
+                dribbler_enabled = self.__parse_dribbler_enabled_event_value(
+                    value=event.value,
+                    max_value=self.controller_config[
+                        RobotControlType.DRIBBLER_ENABLE_2
+                    ][HandheldDeviceConfigKeys.MAX_VALUE],
+                )
+                if dribbler_enabled:
+                    self.motor_control.dribbler_speed_rpm = (
+                        self.dribbler_speed_accumulator
+                    )
+
+        if event.type == ecodes.EV_KEY:
+            if (
+                event.code
+                == self.controller_config[RobotControlType.KICK][
+                    HandheldDeviceConfigKeys.CODE
+                ]
+                and event.value == 1
+            ):
+                self.power_control.geneva_slot = 3
+                self.power_control.chicker.kick_speed_m_per_s = (
+                    self.kick_power_accumulator
+                )
+
+            if (
+                event.code
+                == self.controller_config[RobotControlType.CHIP][
+                    HandheldDeviceConfigKeys.CODE
+                ]
+                and event.value == 1
+            ):
+                self.power_control.geneva_slot = 3
+                self.power_control.chicker.chip_distance_meters = (
+                    self.kick_power_accumulator
+                )
+
+    @staticmethod
+    def __parse_move_event_value(
+        event_value: float, max_value: float, scaling_factor: float
+    ) -> float:
+        if abs(event_value) < (
+            HandheldDeviceConstants.DEADZONE_PERCENTAGE * scaling_factor
+        ):
+            return 0
+        else:
+            return numpy.clip(event_value, 0, max_value * scaling_factor)
+
+    @staticmethod
+    def __parse_dribbler_enabled_event_value(value: float, max_value: float) -> bool:
+        return value > (max_value / 2.0)
+
+    @staticmethod
+    def __parse_dribbler_event_value(value: float) -> float:
+        return numpy.clip(
+            value * HandheldDeviceConstants.DRIBBLER_STEPPER,
+            0,
+            HandheldDeviceConstants.DRIBBLER_INDEFINITE_SPEED,
+        )
+
+    @staticmethod
+    def __parse_kick_event_value(value: float) -> float:
```

Comment:
nit: docs

Reply:
yea I gave up on filling docs for a while now, they are missing pretty much everywhere. Thanks for pointing em out tho for sure, and I'll catch em all eventually lol